runtime/index.d.ts

```diff
@@ -12,16 +19,64 @@ declare module '@sapper/app' {
 }
 
 declare module '@sapper/server' {
-	import { Handler, Req, Res } from '@sapper/internal/manifest-server';
+	import { IncomingMessage, ServerResponse } from 'http';
+	import { TLSSocket } from 'tls';
 
 	export type Ignore = string | RegExp | ((uri: string) => boolean) | Ignore[];
 
+	/**
+	 * The request object passed to middleware and server-side routes. 
+	 * These fields are common to both Polka and Express, but you are free to 
+	 * instead use the typings that come with the server you use.
+	 */
+	export interface SapperRequest extends IncomingMessage {
+		url: string;
+		method: string;	
+		baseUrl: string;
+	
+		/**
+		 * The originally requested URL, including parent router segments.
+		 */
+		originalUrl: string;
+	
+		/**
+		 * The path portion of the requested URL.
+		 */
+		path: string;
+	
+		/**
+		 * The values of named parameters within your route pattern
+		 */
+		params: Record<string, string>;
+	
+		/**
+		 * The un-parsed querystring
+		 */
+		search: string | null;
+	
+		/**
+		 * The parsed querystring
+		 */
+		query: Record<string, string>;
+
+		socket: TLSSocket;
+	}
+
+	export interface SapperResponse extends ServerResponse {
+		locals?: {
+			nonce?: string;
+			name?: string;
+		};
+	}
+		
 	export interface MiddlewareOptions {
-		session?: (req: Req, res: Res) => unknown
-		ignore?: Ignore
+		session?: (req: SapperRequest, res: SapperResponse) => unknown;
+		ignore?: Ignore;
 	}
 
-	export function middleware(opts: MiddlewareOptions): Handler;
+	export function middleware(
+		opts: MiddlewareOptions
```

Comment:
`opts` is still optional, right? Should probably be `opts?`.

Reply:
True. It doesn't really relate to this PR but I'll change it here anyway. Thanks!